packages/@webex/plugin-meetings/src/multistream/sendSlotManager.ts

```diff
@@ -0,0 +1,165 @@
+   * @returns {SendSlot}
+   */
+  public getSlot(mediaType: MediaType): SendSlot {
+    const slot = this.slots.get(mediaType);
+
+    if (!slot) {
+      throw new Error(`Slot for ${mediaType} does not exist`);
+    }
+
+    return slot;
+  }
+
+  /**
+   * This method publishes the given stream to the sendSlot for the given mediaType
+   * @param {MediaType} mediaType MediaType of the sendSlot to which a stream needs to be published (AUDIO_MAIN/VIDEO_MAIN/AUDIO_SLIDES/VIDEO_SLIDES)
+   * @param {LocalStream} stream LocalStream to be published
+   * @returns {Promise<void>}
+   */
+  public async publishStream(mediaType: MediaType, stream: LocalStream): Promise<void> {
+    const slot = this.slots.get(mediaType);
+
+    if (!slot) {
+      throw new Error(`Slot for ${mediaType} does not exist`);
+    }
+
+    await slot.publishStream(stream);
+
+    this.LoggerProxy.logger.info(
+      `SendSlotsManager->publishStream#Published stream for ${mediaType} and stream ${stream}`
+    );
+  }
+
+  /**
+   * This method unpublishes the stream from the sendSlot of the given mediaType
+   * @param {MediaType} mediaType MediaType of the sendSlot from which a stream needs to be unpublished (AUDIO_MAIN/VIDEO_MAIN/AUDIO_SLIDES/VIDEO_SLIDES)
+   * @returns {Promise<void>}
+   */
+  public async unpublishStream(mediaType: MediaType): Promise<void> {
+    const slot = this.slots.get(mediaType);
+
+    if (!slot) {
+      throw new Error(`Slot for ${mediaType} does not exist`);
+    }
+
+    await slot.unpublishStream();
+
+    this.LoggerProxy.logger.info(
+      `SendSlotsManager->unpublishStream#Unpublished stream for ${mediaType}`
+    );
+  }
+
+  /**
+   * This method is used to toggle the active state of the sendSlot for the given mediaType
+   * @param {MediaType} mediaType The MediaType of the sendSlot for which the active state needs to be toggled (AUDIO_MAIN/VIDEO_MAIN/AUDIO_SLIDES/VIDEO_SLIDES)
+   * @param {boolean} active The optional boolean to set the active state of the sendSlot
+   * @returns {void}
+   */
+  public toggleActive(mediaType: MediaType, active?: boolean): void {
+    const slot = this.slots.get(mediaType);
+
+    if (!slot) {
+      throw new Error(`Slot for ${mediaType} does not exist`);
+    }
+
+    if (active === undefined) {
+      active = !slot.active;
+    }
+
+    slot.active = active;
+
+    this.LoggerProxy.logger.info(
+      `SendSlotsManager->toggleActive#Toggled active for ${mediaType} to ${active}`
+    );
+  }
+
+  /**
+   * This method is used to set the codec parameters for the sendSlot of the given mediaType
+   * @param {MediaType} mediaType MediaType of the sendSlot for which the codec parameters needs to be set (AUDIO_MAIN/VIDEO_MAIN/AUDIO_SLIDES/VIDEO_SLIDES)
+   * @param {Object} codecParameters
+   * @returns {Promise<void>}
+   */
+  public async setCodecParameters(
+    mediaType: MediaType,
+    codecParameters: {
+      [key: string]: string | undefined;
```

Comment:
Can we have a key without a value? Is that what this is illustrating?

Reply:
Also, just curious, if we change the codec parameters, shouldn't there also be changes in the stream because of this? Is that automatically handled by the stream?

Reply:
> Can we have a key without a value? Is that what this is illustrating?

Perhaps we should add a comment here to avoid confusion. `undefined` is a valid value used for codec parameters according to ts-sdp.

> Also, just curious, if we change the codec parameters, shouldn't there also be changes in the stream because of this? Is that automatically handled by the stream?

Actually, these are SDP parameters that are being changed. WCME will handle them automatically. (Maybe we need a comment here too? 😅)